tests/test_debian_9.py

```diff
@@ -350,38 +403,180 @@ def test_adhoc_1so_using_default_yml(self):
+            cid = cid.get("Id")
+            self.client.start(cid)
+            # Poll for the container to be ready
+            assert self.wait_for_containers(1, name=splunk_container_name)
+            # Check splunkd
+            splunkd_port = self.client.port(cid, 8089)[0]["HostPort"]
+            url = "https://localhost:{}/services/server/info".format(splunkd_port)
+            kwargs = {"auth": ("admin", password), "verify": False}
+            status, content = self.handle_request_retry("GET", url, kwargs)
+            assert status == 200
+        except Exception as e:
+            self.logger.error(e)
+            raise e
         finally:
             if cid:
-                self.client.remove_container(cid.get("Id"), v=True, force=True)
+                self.client.remove_container(cid, v=True, force=True)
+            try:
+                os.remove(os.path.join(FIXTURES_DIR, "default.yml"))
+            except OSError:
+                pass
+    
+    def test_adhoc_1so_bind_mount_apps(self):
+        # Generate default.yml
+        cid = self.client.create_container(SPLUNK_IMAGE_NAME, tty=True, command="create-defaults")
+        self.client.start(cid.get("Id"))
+        output = self.get_container_logs(cid.get("Id"))
+        self.client.remove_container(cid.get("Id"), v=True, force=True)
+        password = re.search("  password: (.*)", output).group(1).strip()
+        assert password
+        # Write the default.yml to a file
+        with open(os.path.join(FIXTURES_DIR, "default.yml"), "w") as f:
+            f.write(output)
+        # Create the container and mount the default.yml
+        cid = None
+        try:
+            # Spin up this container, but also bind-mount the app in the fixtures directory
+            splunk_container_name = generate_random_string()
+            cid = self.client.create_container(SPLUNK_IMAGE_NAME, tty=True, command="start-service", ports=[8089], 
+                                            volumes=["/tmp/defaults/", "/opt/splunk/etc/apps/splunk_app_example/"], name=splunk_container_name,
+                                            environment={"DEBUG": "true", "SPLUNK_START_ARGS": "--accept-license"},
+                                            host_config=self.client.create_host_config(binds=[FIXTURES_DIR + ":/tmp/defaults/", 
+                                                                                              EXAMPLE_APP + ":/opt/splunk/etc/apps/splunk_app_example/"],
+                                                                                       port_bindings={8089: ("0.0.0.0",)})
+                                            )
+            cid = cid.get("Id")
+            self.client.start(cid)
+            # Poll for the container to be ready
+            assert self.wait_for_containers(1, name=splunk_container_name)
+            # Check splunkd
+            splunkd_port = self.client.port(cid, 8089)[0]["HostPort"]
```

Comment:
It seems like theres a bit of code duplication on this segment, where you verify splunk is up and responding at the /services/server/info path.  Would this be better as its own method, to avoid the duplication?

Reply:
+1, there is already a method that does this but I don't think I call it here and probably should. I'll fix this in my next PR when I add more tests.